### PR TITLE
Modify the templates so they can be built in place.

### LINKED ISSUE #796: Ensure that we don't send diagnostics information during development by default in templates.
Curent .NET Core templates do setup logging and add the stackdriver logger provider for all environments, we should ensure that this only happens for production environments. The development environment should not be sending diagnostics by default.

### PR DESCRIPTION
**Pull request by ILMTitan**
This allows us to edit them in Visual Studio.
This PR also fixes #796.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/TemplateWizards/GoogleProjectTemplateSelectorWizard.cs [hunk 1]
```diff
@@ -128,35 +127,43 @@ public void RunFinished()
...
+                switch (result.SelectedFramework)
+                {
+                    case FrameworkType.NetFramework:
+                        var targetFramework = "net461";
+                        additionalCustomParams.Add($"netcoreapp{result.SelectedVersion.Version}={targetFramework}");
+                        break;
+                    case FrameworkType.NetCore:
+                        // Do nothing.
+                        break;
+                    case FrameworkType.None:
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/TemplateWizards/GoogleProjectTemplateSelectorWizard.cs]: These two cases, `FrameworkType.None` and `default`, seem like the same case.

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.0.csproj [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 
   <PropertyGroup>
-    <TargetFramework>$targetframework$</TargetFramework>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
     <PreserveCompilationContext>true</PreserveCompilationContext>
-    <AssemblyName>$safeprojectname$</AssemblyName>
+    <AssemblyName>SafeProjectName</AssemblyName>
     <OutputType>Exe</OutputType>
-    <PackageId>$safeprojectname$</PackageId>
+    <PackageId>SafeProjectName</PackageId>
```
Comment (ivannaranjo) [inline GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.0.csproj]: Doesn't this hardcode the value of `<PackageId>` to `"SafeProjectName"`? Why removing the placeholders?

File: GoogleCloudExtension/GoogleCloudExtension/TemplateWizards/GoogleProjectTemplateSelectorWizard.cs (hunk 1 above)
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension/TemplateWizards/GoogleProjectTemplateSelectorWizard.cs]: They are different enough that I wanted to provide two different error messages.

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.0.csproj (hunk 2 above)
Comment (ILMTitan) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.0.csproj]: `$safeprojectname$` is a default replacement key, and all default replacement keys are of the form `$keyname$`.  That does not *have* to be true, however. By adding a new replacement key `SafeProjectName` during the `RunStarted()` method of the wizard with the same value, the text `SafeProjectName` gets replaced the same way as `$safeprojectname$` does. The big benefit is that `SafeProjectName` is a legal namespace, allowing us to compile and test our templates in place, without needing to run an experimental instance of visual studio.

**Comment by ILMTitan** [comment]
> How are you replacing the SafeProjectName with the value of $safeprojectid$ when instantiating the templates?

In GoogleCloudExtension/GoogleCloudExtension/TemplateWizards/GoogleProjectTemplateWizard.

    replacements[ReplacementsKeys.EmbeddableSafeProjectNameKey] =
     replacements[ReplacementsKeys.SafeProjectNameKey];

where 

    public const string SafeProjectNameKey = "$safeprojectname$";
    public const string EmbeddableSafeProjectNameKey = "SafeProjectName";

a new replacement is added to the replacements dictionary. The key is `SafeProjectName`, and the value comes from the value of the key `$safeprojectname$`

**Comment by ILMTitan** [comment]
PTAL

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.0.csproj (hunk 2 above)
Comment (ivannaranjo) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.0.csproj]: I see.. Doesn't it also mean that we loose track of which ones are placeholders vs. which symbols are final symbols that should appear on the final template? The mistake I made seems too dangerious vs. the advantage of building the template. I would rather expand the template and build it no?

**Comment by ivannaranjo** [comment]
I'm questioning the basic principle, given the fact that it is very dangerous to have a typo, or even just look through the sources looking for the symbols that are going to be replaced. They look like every other symbol.

Is there away in which we could expand a template on our own while running the tests so we can verify that the templates build instead of this?

**Comment by ivannaranjo** [comment]
Alternatively, if we use a fixed naming pattern that is easy to spot, but still valid C# I would go with that, like `_PlaceHolder_` or something along those lines?

**Comment by ILMTitan** [comment]
I am going to close this for now. I am running into some problems running and building the 1.0 template because of extraneous files that are added when used in place.

**Comment by ILMTitan** [comment]
Actually, I don't care if they compile and run; that is what the ProjectTemplates.Tests project is for. The big win for this is being able to edit files with IntelliSense support.

**Comment by ILMTitan** [comment]
Agreed. Changed the replacement key to `_safe_project_name_`.

**Comment by ILMTitan** [comment]
PTAL

**Pull request merged**